### PR TITLE
Update to truffle 5.1

### PR DESCRIPTION
**Pull request by fleupold**
This PR is a prerequisite for https://github.com/gnosis/mock-contract/pull/20 in that it fixes the web3 API and unit tests to use a more recent truffle version. This way the other PR can only focus on updating the smart contract for solidity v0.6

The main changes are adjusting the unit test to use the web3 API (e.g. `contracts.methods`). This also means we no longer need the `ethereumjs-abi` dependency. I also updated the README.

### Test Plan

CI

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,6 @@
 language: node_js
 node_js:
-- '9'
+- '10'
 cache:
   directories:
     - node_modules
-before_script: 
-- cd node_modules/truffle && npm install solc@0.5.0 && cd ../..
-- truffle compile
```
Comment (bh2smith) [inline .travis.yml:9]: So no more truffle compile?

Also nit, no new line at EOF

File: README.md [hunk 2]
```diff
@@ -90,7 +90,8 @@ it('updates the captor', async () => {
   const mock = await MockContract.new()
   const auction = await SimpleAuction.new(mock.address)
 
-  await mock.givenAnyReturnBool(true)
+  const true_encoded = web3.eth.abi.encodeParameter("bool", true)
```
Comment (bh2smith) [inline README.md:93]: ubernit: JS is usually camelCase.

File: test/MockContract.js [hunk 3]
```diff
@@ -9,9 +8,9 @@ contract('MockContract', function(accounts) {
   describe("cleanState", function() {
     it("should return null if not mocked", async function() {
       const mock = await MockContract.new()
-      const complex = ComplexInterface.at(mock.address)
+      const complex = await ComplexInterface.at(mock.address)
```
Comment (bh2smith) [inline test/MockContract.js:11]: Weird that this wasn't necessary before.

File: .travis.yml (hunk 1 above)
Comment (fleupold) [reply .travis.yml:9]: truffle compile is part of truffle test.

**Pull request merged**